### ISSUE TITLE
Multiple fields in a variant currently fails to compile

### ISSUE THREAD

**Issue by richard-uk1** [Finnerale/druid-enums#11]
Currently, if you have more than 1 field in a variant, the generated code won't compile. You would need to generate a lens to enable everything to hook up correctly. The easiest solution is just to return an error if a variant contains more than 1 field (which is easy enough). The harder solution would be to generate the required lens, but I'm not even sure this is desirable, as it hides some complexity.

The lens would transform `(&mut A, &mut B)` into `&mut (A, B)`. It would do this by taking ownership of `A` and `B` by cloning, allowing the inner methods to modify them, and then replacing the contents of the original tuple with the new values. The cloning should be cheap, because all values used as data should be cheap to clone (either stack-allocated simple types, or types behind reference-counting).

**Comment by luleyleo** [comment]
> Currently, if you have more than 1 field in a variant, the generated code won't compile.

Yup, this was a deliberate decision, as having more than one field does not work with druids `Widget<T>` idea, where one widget represents exactly one type / item.

> The easiest solution is just to return an error if a variant contains more than 1 field (which is easy enough).

Having proper error handling would be good indeed.

> The harder solution would be to generate the required lens, but I'm not even sure this is desirable, as it hides some complexity.

I think the user should explicitly use a tuple inside the enum. This crate already adds enough *magic* as is.

**Comment by richard-uk1** [comment]
OK I agree I think, so I will make this a PR to give a better error message when multiple fields are used.
